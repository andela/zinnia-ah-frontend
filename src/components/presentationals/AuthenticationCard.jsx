/* eslint-disable react/display-name */
import React from 'react';
import { Tab, Form, Image } from 'semantic-ui-react';

import Button from '../presentationals/Button';
import '../presentationals/AuthenticationCard.scss';
import '../presentationals/Form.scss';
import GoogleIcon from '../../assets/images/google-icon.svg';
import FacebookIcon from '../../assets/images/facebook-icon.svg';
import TwitterIcon from '../../assets/images/twitter-icon.svg';
import Title from './Title';
import Signup from '../signup/Signup';

const panes = [
  {
    menuItem: 'Sign Up',
    render: () => (
      <Tab.Pane attached={false}>
        <Signup />
      </Tab.Pane>
    ),
  },
  {
    menuItem: 'Login',
    render: () => (
      <Tab.Pane attached={false}>
        <Form>
          <Form.Field>
            <input placeholder="Username or Email" />
          </Form.Field>
          <Form.Field>
            <input placeholder="Password" type="password" />
          </Form.Field>
          <div className="text-right">
            <a href="/forgot-password" className="text-red">
              Forgot Password?
            </a>
          </div>
          <Button button={loginButtonValues} />
        </Form>
        <div className="d-flex or-div">
          <hr />
          <p className="or">OR</p>
          <hr />
        </div>
        <div className="icon-container">
          <div className="icon-card">
            <Image src={GoogleIcon} />
          </div>
          <div className="icon-card">
            <Image src={FacebookIcon} />
          </div>
          <div className="icon-card">
            <Image src={TwitterIcon} />
          </div>
        </div>
      </Tab.Pane>
    ),
  },
];

const AuthenticationCard = () => {
  return (
    <div className="right">
      <div className="card">
        <Title title={title} />
        <hr />
        <Tab
          className="tab-label"
          menu={{ secondary: true, pointing: true }}
          panes={panes}
        />
      </div>
    </div>
  );
};

const title = {
  content: 'Become an Author',
  class: 'text-center',
};

const loginButtonValues = {
  type: 'submit',
  value: 'CONTINUE',
  class: 'btn-dark',
};

export default AuthenticationCard;
