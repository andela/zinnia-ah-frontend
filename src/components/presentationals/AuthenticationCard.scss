@import "../../assets/variables";

.card {
  width: 467px;
  padding: 30px 48px;
  background: #FFFFFF;
  border: 1px solid rgba(0, 0, 0, 0.25);
  box-sizing: border-box;
  box-shadow: 0 15px 90px rgba(0, 0, 0, 0.1);
  border-radius: 10px;
  margin-top: 30px;
}

hr {
  width: 107px;
  background-color: $secondary-grey;
  height: 1px;
  border: none;
  margin-bottom: 24px;
}

.tab-label {

  .menu {
    border-bottom: 2px solid $secondary-grey;

    .item {
      color: $primary-grey;
      width: 50%;
    }
  }

  .segment {
    padding: 0;
    border: none;
    box-shadow: none;
    -webkit-box-shadow: none;
  }
}

.right {

  .or-div {
    height: 22px;
    display: flex;
    align-items: center;
    margin-top: 25px;

    hr {
      width: 163px !important;
      margin-top: 20px;
    }

    .or {
      font-weight: bold;
      font-size: 18px;
      line-height: 22px;
      text-align: center;
      color: #000000;
      margin: 0 5px;
    }
  }

  .icon-container {
    display: flex;
    justify-content: space-around;
    margin-top: 20px;
    margin-bottom: 40px;

    .icon-card {
      width: 70px;
      height: 70px;
      background: #FFFFFF;
      box-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
      border-radius: 5px;
      display: flex;
      justify-content: center;
      align-items: center;
    }
  }
}

.text-right {
  text-align: right;
}

.text-red {
  color: $primary-red;
}

.forgot-password {
  margin-top: 175px;
}

.reset-password {
  margin-top: 150px;
}
