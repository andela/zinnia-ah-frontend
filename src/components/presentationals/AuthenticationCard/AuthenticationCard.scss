@import '../../../assets/variables';

.auth-card {
  background: $primary-white;
  border: .0625rem solid $primary-black-opacity-2;
  border-radius: .625rem;
  box-shadow: 0 .9375rem 5.625rem $primary-black-opacity-1;
  box-sizing: border-box;
  margin-top: 1.875rem;
  padding: 1.875rem 3rem;
  width: 29.1875rem;
}

.hr {
  background-color: $secondary-grey;
  border: 0;
  height: .0625rem;
  margin-bottom: 1.5rem;
  width: 6.6875rem;
}

.tab-label {
  .menu {
    border-bottom: .125rem solid $secondary-grey;

    .item {
      color: $primary-grey;
      width: 50%;
    }
  }

  .segment {
    border: 0;
    box-shadow: none;
    padding: 0;
  }
}

.right {
  .or-div {
    align-items: center;
    display: flex;
    height: 1.375rem;
    margin-top: 1.5625rem;

    .hr {
      margin-top: 1.25rem;
      width: 10.1875rem;
    }

    .or {
      color: $primary-black;
      font-size: 1.125rem;
      font-weight: bold;
      line-height: 1.375rem;
      margin: 0 .3125rem;
      text-align: center;
    }
  }

  .icon-container {
    display: flex;
    justify-content: space-around;
    margin-bottom: 2.5rem;
    margin-top: 1.25rem;

    .icon-card {
      align-items: center;
      background: $primary-white;
      border-radius: .3125rem;
      box-shadow: 0 .25rem .25rem $primary-black-opacity-2;
      display: flex;
      height: 4.375rem;
      justify-content: center;
      width: 4.375rem;
    }
  }

  .text-right {
    text-align: right;
  }

  .text-red {
    color: $primary-red;
  }

  .forgot-password {
    margin-top: 10.9375rem;
  }

  .reset-password {
    margin-top: 9.375rem;
  }
}
