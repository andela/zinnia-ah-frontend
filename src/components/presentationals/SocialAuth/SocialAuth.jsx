import React from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import qs from 'qs';

//modules
import {
  loginInitialize,
  socialSuccess,
  loginError,
} from '../../../store/modules/auth';

// components
import Loader from '../Loader/Loader';

//helpers
import { setToken, decodeToken } from '../../../api/helpers';

const SocialAuth = ({ location, history, ...props }) => {
  props.loginInitialize();

  const loader = (
    <Loader data-test="loader" size="large" text="redirecting, please wait" />
  );

  try {
    let query = location.search;
    query = query.replace('?', '');
    const { token, isNewRecord } = qs.parse(query);
    const { user } = decodeToken(token);
    const { username, email, id } = user;
    let message;

    isNewRecord === 'true'
      ? (message = `Hi, ${email}, Welcome to Authors Haven`)
      : (message = `Welcome Back ${username}`);

    history.push('/');
    toast.success(message);
    setToken(token);
    props.socialSuccess({ email, username, id });

    return loader;
  } catch (error) {
    const message = 'Authentication failed. Please try again';
    history.push('/login');
    toast.error(error.message);
    props.loginError([message]);
    return loader;
  }
};

export default connect(
  null,
  { loginInitialize, socialSuccess, loginError },
)(SocialAuth);
