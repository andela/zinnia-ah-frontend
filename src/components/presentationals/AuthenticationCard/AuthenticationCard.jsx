import React from 'react';
import { Tab } from 'semantic-ui-react';

// components
import Button from '../Button/Button';
import Image from '../Image/Image';
import Title from '../../presentationals/Title/Title';
import SignupForm from '../../Register/Register';

// styles
import './AuthenticationCard.scss';
import '../../presentationals/Form.scss';

// images
import GoogleIcon from '../../../assets/images/google-icon.svg';
import FacebookIcon from '../../../assets/images/facebook-icon.svg';
import TwitterIcon from '../../../assets/images/twitter-icon.svg';

const panes = [
  {
    menuItem: 'Sign Up',
    render: () => (
      <Tab.Pane attached={false}>
        <SignupForm />
      </Tab.Pane>
    ),
  },
  {
    menuItem: 'Login',
    render: () => (
      <Tab.Pane attached={false}>
        <form className="form">
          <div className="form-group">
            <input className="form-control" placeholder="Username or Email" />
          </div>
          <div className="form-group">
            <input
              className="form-control"
              placeholder="Password"
              type="password"
            />
          </div>
          <div className="text-right">
            <a href="/forgot-password" className="text-red">
              Forgot Password?
            </a>
          </div>
          <Button type="submit" value="CONTINUE" className="btn-dark" />
        </form>
        <div className="d-flex or-div">
          <hr className="hr" />
          <p className="or">OR</p>
          <hr className="hr" />
        </div>
        <div className="icon-container">
          <div className="icon-card">
            <Image src={GoogleIcon} />
          </div>
          <div className="icon-card">
            <Image src={FacebookIcon} />
          </div>
          <div className="icon-card">
            <Image src={TwitterIcon} />
          </div>
        </div>
      </Tab.Pane>
    ),
  },
];

const AuthenticationCard = () => {
  return (
    <div className="right">
      <div className="auth-card">
        <Title content="Become an Author" className="text-center" />
        <hr className="hr" />
        <Tab
          className="tab-label"
          menu={{ secondary: true, pointing: true }}
          panes={panes}
        />
      </div>
    </div>
  );
};

export default AuthenticationCard;
