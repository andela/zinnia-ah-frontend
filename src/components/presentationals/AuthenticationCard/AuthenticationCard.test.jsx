import React from 'react';
import { shallow } from 'enzyme';
import checkPropTypes from 'check-prop-types';

import AuthenticationCard from '../AuthenticationCard/AuthenticationCard';

import { findByTestAttribute } from '../../../components/presentationals/SocialAuth/SocialAuth.test';

const setupComponent = (props = {}) => {
  const component = shallow(<AuthenticationCard {...props} />);
  return component;
};

describe('<Authentication Card />', () => {
  let component;

  beforeEach(() => {
    component = setupComponent({ pathname: 'example prop' });
  });

  it('should render with no errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('should receive props', () => {
    const expectedProps = { pathname: 'example prop' };

    const propsError = checkPropTypes(
      AuthenticationCard.propTypes,
      expectedProps,
      'props',
      AuthenticationCard.pathname,
    );

    expect(propsError).toBeUndefined();
  });

  it('should render the social icons', () => {
    const wrapper = findByTestAttribute(component, 'social-icon');
    expect(wrapper.length).toBe(3);
    expect(wrapper.children().length).toBe(3);
  });
});
