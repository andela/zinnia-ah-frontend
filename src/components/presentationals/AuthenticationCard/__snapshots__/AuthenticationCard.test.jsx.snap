// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication Card /> should render with no errors 1`] = `
<div
  className="right"
>
  <div
    className="auth-card"
  >
    <Title
      className="text-center"
      content="Become an Author"
    />
    <hr
      className="hr"
    />
    <div
      className="tab-label"
    >
      <div
        className="ui pointing secondary menu"
      >
        <Link
          className="item"
          to="/signup"
        >
          Sign Up
        </Link>
        <Link
          className="item"
          to="/login"
        >
          Login
        </Link>
      </div>
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Login)",
              "type": [Function],
            }
          }
          path="/login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Register)",
              "type": [Function],
            }
          }
          path="/signup"
        />
      </Switch>
      <div
        className="d-flex or-div"
      >
        <hr />
        <p
          className="or"
        >
          OR
        </p>
        <hr />
      </div>
      <div
        className="icon-container"
      >
        <a
          href="http://localhost:3000/api/v1/auth/google?redirectTo=https://zinnia-ah-frontend-staging.herokuapp.com/social-auth"
        >
          <div
            className="icon-card"
            data-test="social-icon"
          >
            <Image
              as="img"
              src=""
              ui={true}
            />
          </div>
        </a>
        <a
          href="http://localhost:3000/api/v1/auth/facebook?redirectTo=https://zinnia-ah-frontend-staging.herokuapp.com/social-auth"
        >
          <div
            className="icon-card"
            data-test="social-icon"
          >
            <Image
              as="img"
              src=""
              ui={true}
            />
          </div>
        </a>
        <a
          href="http://localhost:3000/api/v1/auth/twitter?redirectTo=https://zinnia-ah-frontend-staging.herokuapp.com/social-auth"
        >
          <div
            className="icon-card"
            data-test="social-icon"
          >
            <Image
              as="img"
              src=""
              ui={true}
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
